### ISSUE TITLE
deviceId 

### ISSUE THREAD

**Issue by MaYinjian** [alexa/skill-sample-node-device-address-api#16]
how to generate a unique deviceId? We used UUID to find that it was repeated  in  android

**Comment by franklin-lobb** [comment]
The deviceId is provided as part of the payload sent to your skill, so you do not need to generate it.

**Comment by MaYinjian** [comment]
Yes, I do not need to generate it，but what i care is where the deviceId saved， is it save in file systems? external storage or internal storage like data/data  ? We are concerned that the user will be cleared when they restore the factory Settings.Once deviceID is refreshed, it's considered a new device.
